lib/ex_unit/lib/ex_unit/doc_test.ex

```diff
@@ -208,6 +211,7 @@ defmodule ExUnit.DocTest do
 
         for {name, test} <- ExUnit.DocTest.__doctests__(module, opts) do
           @file file
+          @tag Keyword.get(opts, :tags, [])
```

Comment:
This is a micro optimization but I think we should do this:

```
tags =
  if tags = Keyword.get(opts, :tags, []) do
    quote do: @tag unquote(tags)
  end
```

and then in here you do:

```
unquote(tags)
```

So we don't end-up calling the module attribute with nothing most of the times.

Reply:
I tried pulling it out like that, but because of the `bind_quoted` I couldn't unquote in this `quote` block, and then of course I couldn't use `bind_quoted` with the AST for `@tag unquote(tags)` because the binding was inserting that AST in the resulting output in the wrong place, so I ended up going with this in the quote block with the already bound `opts`:

```
          if tags = Keyword.get(opts, :tags) do
            @tag tags
          end
```

That way it doesn't call the module attribute every time, but it still works.

Let me know if there's a better way to handle this sort of thing!